tee-worker/core/tls-websocket-server/src/connection.rs

```diff
@@ -127,20 +127,49 @@ where
 	///
 	/// Returns a boolean 'connection should be closed'.
 	fn read_or_initialize_websocket(&mut self) -> WebSocketResult<bool> {
-		if let StreamState::EstablishedWebsocket(web_socket) = &mut self.stream_state {
+		if let (StreamState::EstablishedWebsocket(web_socket), _, is_inited)
+		| (StreamState::AllReady(web_socket, ..), is_inited, _) = (&mut self.stream_state, true, false)
+		{
 			trace!(
 				"Read is possible for connection {}: {}",
 				self.connection_token.0,
 				web_socket.can_read()
 			);
 			match web_socket.read_message() {
-				Ok(m) =>
+				Ok(Message::Binary(key)) if !is_inited => {
+					debug!("server reads binary message as after established, encryption enabled");
+					let mut state = StreamState::Invalid;
+					std::mem::swap(&mut self.stream_state, &mut state);
```

Comment:
Why can't we just use 
```
std::mem::replace(&mut self.stream_state, StreamState::Invalid) 
```
Instead of having to initialise `state` and then swapping with it. I'm just curious although it does the same thing, i feel `replace` is more idiomatic.